app/user/rest.py

```diff
@@ -474,13 +480,14 @@ def set_permissions(user_id, service_id):
 @user_blueprint.route("/email", methods=["POST"])
 def fetch_user_by_email():
     email = email_data_request_schema.load(request.get_json())
-
     try:
         fetched_user = get_user_by_email(email["email"])
     except Exception:
+        check_throttle_for_requester(email["email"])
+        add_failed_login_for_requester()
         log_auth_activity(email["email"], "Attempted Login", admin_only=False)
         raise
-
+    check_throttle_for_requester(fetched_user)
```

Comment:
Why have 2 instances of `check_throttle_for_requester`, you could just check it once before the `try`